src/plugins/shape-generator/index.ts

```diff
@@ -0,0 +1,407 @@
+          const exprIdShapeGenerator = exprId?.match(
+            /^shape-generator-(\w+)(?:-|$)/
+          )?.[1];
+
+          if (!exprIdShapeGenerator) {
+            continue;
+          }
+
+          const newDropdownItem = this.newDropdownItems.find(
+            (item) => item.label === exprIdShapeGenerator
+          );
+
+          if (!newDropdownItem) {
+            continue;
+          }
+
+          const expressionTab = document.querySelector(
+            `.dcg-expressionitem[expr-id=${JSON.stringify(
+              newDropdownItem.okBtnExprId
+            )}] .dcg-tab-interior`
+          );
+
+          if (!expressionTab) {
+            continue;
+          }
+
+          if (expressionTab.querySelector(".shape-generator-ok-btn")) {
+            continue;
+          }
+
+          const okButton = document.createElement("button");
+          okButton.id = `shape-generator-${newDropdownItem.label}-ok-btn`;
+          okButton.classList.add("shape-generator-ok-btn");
+          okButton.textContent = "OK";
+          okButton.addEventListener("click", () => {
+            const expr = this.calc
+              .getExpressions()
+              .find((expr) => expr.id === newDropdownItem.okBtnExprId);
+
+            if (!expr) {
+              throw new Error(
+                "Could not find the expression for the OK button"
+              );
+            }
+
+            newDropdownItem.okBtnHandler(expr);
+          });
+
+          expressionTab.appendChild(okButton);
+        }
+      }
+    });
+    const expressionlist = document.querySelector(
+      ".dcg-expressionitem[expr-id]"
+    )?.parentElement;
+    if (!expressionlist) {
+      throw new Error("Could not find the expression list");
+    }
+    this.expressionsMutationObserver.observe(expressionlist, {
+      childList: true,
+    });
+
+    this.afterConfigChange();
+  }
+
+  afterDisable() {
+    // Remove stylesheet
+    document.getElementById("shape-generator-styles")!.remove();
+
+    // Remove global dataset attributes
+    delete document.body.dataset.shapeGeneratorIsEditingShape;
+    delete document.body.dataset.shapeGeneratorShowSliders;
+
+    // Remove event listeners
+    this.calc.controller.dispatcher.unregister(this._onDispatchHandler);
+    this.expressionsMutationObserver?.disconnect();
+    this.expressionsMutationObserver = null;
+
+    // Remove generated expressions
+    this.cleanupExpressions();
+  }
+
+  afterConfigChange() {
+    document.body.dataset.shapeGeneratorShowSliders =
+      this.settings.showSliders.toString();
+  }
+
+  cleanupExpressions() {
+    this.calc.removeExpressions(ellipseGeneratorExpressions);
+    this.calc.removeExpressions(rectangleGeneratorExpressions);
+  }
+
+  get isEditingShape() {
+    return this._isEditingShape;
+  }
+
+  set isEditingShape(value) {
+    this._isEditingShape = value;
+
+    document.body.dataset.shapeGeneratorIsEditingShape = value.toString();
```

Comment:
Since this is just a boolean, a CSS class feels more appropriate. We have precedent with `"dsm-better-nav-scrollable-expressions"`. But I doubt it's worth changing it. Both implementation approaches behave about the same.